xl_auth/templates/collections/view.html

```diff
@@ -32,62 +32,80 @@ <h1>{{ _('View Collection \'%(code)s\'', code=collection.code) }}</h1>
             <dd>
-                {{ collection.modified_at.strftime('%Y-%m-%dT%H:%M:%SZ') }}, {{  _('by') }}
-                <a href="{{ url_for('user.view', user_id=collection.modified_by.id) }}">
-                    {{ collection.modified_by.full_name }}
-                </a>
+                {% for permission in collection.permissions | selectattr('cataloging_admin')
+                        | sort(attribute='created_at') %}
+                    <a href="{{ url_for('user.view', user_id=permission.user.id) }}">{{
+                        permission.user.email }}</a>{{
+                            _(' and ') if loop.revindex == 2 else (', ' if not loop.last) }}
+                {% else %}
+                    <em>{{ _('None registered, please contact libris@kb.se to create one.') }}</em>
+                {% endfor %}
             </dd>
 
-            <dt>{{ _('Created At') }}:</dt>
-            <dd>
-                {{ collection.created_at.strftime('%Y-%m-%dT%H:%M:%SZ') }}, {{  _('by') }}
-                <a href="{{ url_for('user.view', user_id=collection.created_by.id) }}">
-                    {{ collection.created_by.full_name }}
-                </a>
-            </dd>
+            {% if current_user.is_admin or current_user.is_cataloging_admin %}
+                <dt>{{ _('Last Modified') }}:</dt>
+                <dd>
+                    {{ collection.modified_at.strftime('%Y-%m-%dT%H:%M:%SZ') }}, {{  _('by') }}
+                    <a href="{{ url_for('user.view', user_id=collection.modified_by.id) }}">
+                        {{ collection.modified_by.full_name }}
+                    </a>
+                </dd>
+
+                <dt>{{ _('Created At') }}:</dt>
+                <dd>
+                    {{ collection.created_at.strftime('%Y-%m-%dT%H:%M:%SZ') }}, {{  _('by') }}
+                    <a href="{{ url_for('user.view', user_id=collection.created_by.id) }}">
+                        {{ collection.created_by.full_name }}
+                    </a>
+                </dd>
+            {% endif %}
         </dl>
     </div>
-    <div class="panel panel-default">
-        <div class="panel-heading">
-            <strong>{{ _('Permissions') }}</strong>
-        </div>
-        <table class="table table-hover">
-            <thead>
-            <tr>
-                <th class="col-md-2">{{ _('User') }}</th>
-                <th class="col-md-3 word-break-all">{{ _('Registrant') }}</th>
-                <th class="col-md-2 word-break-all">{{ _('Cataloger') }}</th>
-                <th class="col-md-3 word-break-all">{{ _('Cataloguing Administrator') }}</th>
-                <th class="col-md-2">{{ _('Created At') }}</th>
-            </tr>
-            </thead>
-            <tbody>
-            {% for permission in collection.permissions | sort(attribute='created_at') %}
+    {% if current_user.is_admin or current_user.is_cataloging_admin
+            or current_user.has_any_permission_for(collection) %}
```

Comment:
I'm trying to figure this out. For a regular user looking at a collection, they will only see the Permissions table if they have at least one permission for that collection? And if that's the case, they will only see their own permissions (and those of the cataloging admins?)?

For the user view, I removed the bottom table in `simple_view.html` (which is shown to regular users). What do you think about doing the same for this? Regular users can see their own permissions by looking at their profile, so maybe we don't need to show them here too?

Reply:
> I'm trying to figure this out. For a regular user looking at a collection, they will only see the Permissions table if they have at least one permission for that collection? And if that's the case, they will only see their own permissions (and those of the cataloging admins?)?

Exactly.

> For the user view, I removed the bottom table in simple_view.html (which is shown to regular users). What do you think about doing the same for this? /.../

When viewing yourself, you see your own permissions. I think it makes sense to see the permissions for your "own" collections too. Or maybe it doesn't, but let's collect feedback from UX on this release and see what they think!